protocol/protocol.re

```diff
@@ -60,26 +60,35 @@ let apply_side_chain = (state: t, operation) => {
   let included_operations = Set.add(operation, state.included_operations);
 
   let {source, amount, ticket, _} = operation;
-  let.ok (ledger, result) =
-    switch (operation.kind) {
-    | Transaction({destination}) =>
-      let.ok ledger =
-        Ledger.transfer(~source, ~destination, amount, ticket, state.ledger);
-      Ok((ledger, `Transaction));
-    | Withdraw({owner}) =>
-      let.ok (ledger, handle) =
-        Ledger.withdraw(
-          ~source,
-          ~destination=owner,
-          amount,
-          ticket,
-          state.ledger,
-        );
-      // TODO: publish the handle somewhere
-      Ok((ledger, `Withdraw(handle)));
-    };
-
-  Ok(({...state, ledger, included_operations}, result));
+  switch (operation.kind) {
+  | Transaction({destination}) =>
+    let.ok ledger =
+      Ledger.transfer(~source, ~destination, amount, ticket, state.ledger);
+    Ok(({...state, ledger, included_operations}, `Transaction));
+  | Withdraw({owner}) =>
+    let.ok (ledger, handle) =
+      Ledger.withdraw(
+        ~source,
+        ~destination=owner,
+        amount,
+        ticket,
+        state.ledger,
+      );
+    // TODO: publish the handle somewhere
+    Ok(({...state, ledger, included_operations}, `Withdraw(handle)));
+  | Add_validator(validator) =>
+    let validators = Validators.add(validator, state.validators);
+    Ok((
+      {...state, validators, included_operations},
+      `Add_validator(validator),
```

Comment:
Having `validator` parameter in the polyvar `Add_validator` doesn't seem useful. I think we can get rid of it

Reply:
Agreed, if it is useful in the feature we can add it back.

Reply:
Addressed in the next commit